renderdoc/driver/ihv/amd/amd_counters.cpp

```diff
@@ -22,70 +22,202 @@
  * THE SOFTWARE.
  ******************************************************************************/
 
+#include <sstream>
+#include <iostream>
+#include <chrono>
+#include <thread>
 #include "amd_counters.h"
-#include <Windows.h>
 #include "common/common.h"
 #include "core/plugins.h"
 #include "official/GPUPerfAPI/Include/GPUPerfAPI.h"
 #include "official/GPUPerfAPI/Include/GPUPerfAPIFunctionTypes.h"
-
 #include "strings/string_utils.h"
 
+#if ENABLED(RDOC_WIN32)
+#include <Windows.h>
+#else
+#include <dlfcn.h>
+#endif
+
+inline bool AMD_FAILED(GPA_Status status)
+{
+  return status != GPA_STATUS_OK;
+}
+
+inline bool AMD_SUCCEEDED(GPA_Status status)
+{
+  return status == GPA_STATUS_OK;
+}
+
+static void GPA_LoggingCallback(GPA_Logging_Type messageType, const char* pMessage)
+{
+  std::string message;
+
+  switch (messageType)
+  {
+  case GPA_LOGGING_ERROR:
+    message = "Error: ";
+    break;
+  case GPA_LOGGING_MESSAGE:
+    message = "Info: ";
+    break;
+  case GPA_LOGGING_TRACE:
+    message = "Trace: ";
+    break;
+  default:
+    break;
+  }
+
+  message += "AMD: ";
```

Comment:
Similarly I don't think there's a need for an extra prefix, this will already be noted as coming from `amd_counters.cpp`.

Reply:
Fixed